src/access/workflows/MorphoWorkflow.sol

```diff
@@ -0,0 +1,140 @@
+    address public constant LOAN_TOKEN = 0xaf88d065e77c8cC2239327C5EDb3A432268e5831;
+
+    /// @notice Collateral token ($K on Arbitrum)
+    address public constant COLLATERAL_TOKEN = 0x010700AB046Dd8e92b0e3587842080Df36364ed3;
+
+    /// @notice Oracle address
+    address public constant ORACLE = 0x2964aB84637d4c3CAF0Fd968be1c97D9990de925;
+
+    /// @notice Interest Rate Model
+    address public constant IRM = 0x66F30587FB8D4206918deb78ecA7d5eBbafD06DA;
+
+    /// @notice Liquidation Loan-to-Value (62.5%)
+    uint256 public constant LLTV = 625000000000000000;
+
+    /* ============ Internal Functions ============ */
+
+    /**
+     * @notice Creates the MarketParams struct for Morpho operations
+     * @return The MarketParams struct for Morpho operations
+     */
+    function _getMarketParams() internal pure returns (MarketParams memory) {
+        return MarketParams({
+            loanToken: LOAN_TOKEN,
+            collateralToken: COLLATERAL_TOKEN,
+            oracle: ORACLE,
+            irm: IRM,
+            lltv: LLTV
+        });
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * @notice Supplies collateral and optionally borrows loan tokens
+     * @param amountLend The amount of collateral tokens to lend
+     * @param amountBorrow The amount of loan tokens to borrow (can be 0)
+     * @return borrowed The amount of loan tokens borrowed
+     */
+    function lendAndBorrow(uint256 amountLend, uint256 amountBorrow) external returns (uint256 borrowed) {
+        // Get market params
+        MarketParams memory marketParams = _getMarketParams();
+
+        // Approve Morpho to spend collateral tokens
+        IERC20(COLLATERAL_TOKEN).forceApprove(MORPHO, amountLend);
+
+        // Supply collateral to Morpho
+        IMorpho(MORPHO).supplyCollateral(marketParams, amountLend, address(this), "");
+
+        // If amountBorrow > 0, borrow loan tokens
+        if (amountBorrow > 0) {
+            // Borrow loan tokens from Morpho
+            (borrowed,) = IMorpho(MORPHO).borrow(marketParams, amountBorrow, 0, address(this), address(this));
+        }
+
+        return borrowed;
+    }
+
+    /**
+     * @notice Repays loan and optionally withdraws collateral
+     * @param amountRepay The amount of loan tokens to repay (can be 0)
+     * @param amountWithdraw The amount of collateral tokens to withdraw
+     * @return withdrawn The amount of collateral tokens withdrawn
+     */
+    function repayAndWithdraw(uint256 amountRepay, uint256 amountWithdraw) external returns (uint256 withdrawn) {
+        // Get market params
+        MarketParams memory marketParams = _getMarketParams();
+
+        // If amountRepay > 0, repay loan
+        if (amountRepay > 0) {
+            // Approve Morpho to spend loan tokens
+            IERC20(LOAN_TOKEN).forceApprove(MORPHO, amountRepay);
+
+            // Repay loan to Morpho
+            IMorpho(MORPHO).repay(marketParams, amountRepay, 0, address(this), "");
```

Comment:
what happens if you try to repay a few more gwei than you have? Does it revert?

Reply:
It doesn't revert and the money are donated to the Morpho.
<img width="896" alt="image" src="https://github.com/user-attachments/assets/a66eb867-5cf8-4b4d-b6a9-f0c8fd8a2c47" />